### PR TITLE
[Improvement][alert-spi]plugin instance only saves the main information

### PR DESCRIPTION
**Pull request by CalvinKirs**
plugin instance only saves the main information

when users need to display all the complete information (usually UI display), then do the conversion.

Store plug-in information when loading plug-in, query database when removing alert

### CODE REVIEW

**Comment by felix-thinkingdata** [comment]
Let me verify that

**Review by felix-thinkingdata — APPROVED** [review]
LGTM

**Review by felix-thinkingdata — APPROVED** [review]

File: dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/runner/AlertSender.java [hunk 1]
```diff
@@ -157,7 +155,8 @@ private AlertResult alertResultHandler(AlertPluginInstance instance, AlertData a
 
         AlertInfo alertInfo = new AlertInfo();
         alertInfo.setAlertData(alertData);
-        alertInfo.setAlertParams(instance.getPluginInstanceParams());
+        Map<String, String> paramsMap = JSONUtils.toMap(instance.getPluginInstanceParams());
+        alertInfo.setAlertParams(paramsMap);
         AlertResult alertResult = alertChannel.process(alertInfo);
```
Comment (felix-thinkingdata) [inline dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/runner/AlertSender.java:160]: When a message needs to be sent by more than one alert instance, one of the alert instances will throw an exception here, causing the other instance alerts to be affected.
一个消息，需要多个告警实例发送时，其中一个告警实例此处抛出异常，会造成其他实例告警受影响。
This is also possible if the alert plug-in catches exceptions.
如果告警插件捕获异常也是可以的。

**Review by felix-thinkingdata — APPROVED** [review]
LGTM 👍

**Review by gaojun2048 — APPROVED** [review]
+1

**Pull request merged**